src/assets/src/components/queueSettings.tsx

```diff
@@ -0,0 +1,315 @@
+                            <div className='mt-4'>
+                                <Button variant='danger' disabled={props.disabled} aria-label='Delete Queue' onClick={props.onDeleteClick}>
+                                    Delete Queue
+                                </Button>
+                            </div>
+                        </Tab.Pane>
+                    </Tab.Content>
+                </Col>
+            </Row>
+        </Tab.Container>
+    );
+}
+
+interface SettingsPageParams {
+    queue_id: string;
+}
+
+export function ManageQueueSettingsPage(props: PageProps<SettingsPageParams>) {
+    if (!props.user) {
+        redirectToLogin(props.loginUrl);
+    }
+
+    // Set up page state
+    const dialogRef = createRef<Dialog>();
+    const [queue, setQueue] = useState(undefined as QueueHost | undefined);
+    const [queueInitialized, setQueueInitialized] = useState(false);
+    const [authError, setAuthError] = useState(undefined as Error | undefined);
+
+    // Set up WebSocket
+    const queueID = props.match.params.queue_id;
+    if (queueID === undefined) throw new Error("queueID is undefined!");
+    if (!props.user) throw new Error("user is undefined!");
+    const queueIDInt = Number(queueID);
+
+    const setQueueChecked = (q: QueueAttendee | QueueHost | undefined) => {
+        if (!q) {
+            setQueue(q);
+        } else if (isQueueHost(q)) {
+            setQueue(q);
+            setAuthError(undefined);
+        } else {
+            setQueue(undefined);
+            setAuthError(new Error("You are not a host of this queue. If you believe you are seeing this message in error, contact the queue host(s)."));
+        }
+    }
+    const userWebSocketError = useQueueWebSocket(queueIDInt, setQueueChecked);
+
+    const [activeKey, setActiveKey] = useState(AvailableTabs.General as AvailableTabs);
+    const [showCorrectGeneralMessage, setShowCorrectGeneralMessage] = useState(false);
+    const [showSuccessMessage, setShowSuccessMessage] = useState(false);
+    const [name, setName] = useState('');
+    const [nameValidationResult, setNameValidationResult] = useState(undefined as undefined | ValidationResult);
+    const [description, setDescription] = useState('');
+    const [descriptValidationResult, setDescriptValidationResult] = useState(undefined as undefined | ValidationResult);
+    const [allowedMeetingTypes, setAllowedMeetingTypes] = useState(new Set() as Set<string>);
+    const [allowedValidationResult, setAllowedValidationResult] = useState(undefined as undefined | MeetingTypesValidationResult);
+
+    if (queue && !queueInitialized) {
+        setQueueInitialized(true);
+        setName(queue.name);
+        setDescription(queue.description);
+        setAllowedMeetingTypes(new Set(queue.allowed_backends));
+    }
+
+    const resetValidationResults = () => {
+        if (setShowCorrectGeneralMessage) setShowCorrectGeneralMessage(false);
```

Comment:
So, I could remove the `if` for simplicity, but it made sense to me to only update the state if it needed to be. It's possible the user doesn't enter anything invalid and then hits submit, and this would still be `false`. I'm happy to change it if you'd prefer it the other way.

Reply:
I think that React only "does" anything (re-renders) if the value you set is different than the previous value, so I think calling the setter is a no-op if it's already `false`. But I could be wrong on this, and this is absolutely a quibble so feel free to resolve & ignore!

Reply:
Huh, if that's true, I'm happy to simplify.

Reply:
Done. https://github.com/tl-its-umich-edu/remote-office-hours-queue/pull/173/commits/7c3baf7b84495f780bb8730d544bf9b1d917e3ff